src/contrib/electron_motion/validateNUE.C

```diff
@@ -0,0 +1,335 @@
+  TH1F *vz_dist = new TH1F("vz",char_title_car,20,-0.15,0.15);
+
+  //2D histogram with cos theta and phi of velocities
+  TH2F *ang_dist = new TH2F("vxy",char_title_ang,20,-1,1,20,-3.1415,3.1415);
+
+  NtpMCEventRecord * mcrec = 0;
+  tree->SetBranchAddress( "gmcrec", &mcrec);
+
+  // Event loop
+  for(Long64_t i=0; i < tree->GetEntries(); i++){
+    tree->GetEntry(i);
+
+    EventRecord & event = *( mcrec->event );
+    
+    const Interaction & inter = *( event.Summary() ) ;
+
+    const ProcessInfo & proc_info = inter.ProcInfo() ;
+
+    if (proc_info.IsElectronScattering()){
+      GHepParticle * electron = event.HitElectron(); //Find initial electron
+      const TLorentzVector & pe = *( electron -> P4() ) ; //Get its 4 momentum
+      //Get velocity from four momentum
+      //double vx = calc_velocity(pe.Px(),electron->Mass());
+      double vx = pe.Px()/electron->Energy();
+      double vy = pe.Py()/electron->Energy();
+      double vz = pe.Pz()/electron->Energy();
+
+      //Get angles
+      double costheta = vz/(sqrt(pow(vx,2)+pow(vy,2)+pow(vz,2)));
+      double phi = vy/abs(vy)*acos(vx/(sqrt(pow(vx,2)+pow(vy,2))));
+
+      //Fill histograms
+      vx_dist->Fill(pe.Px());
+      vy_dist->Fill(pe.Py());
+      vz_dist->Fill(pe.Pz());
+
+      //Fill 2d histogram with angles
+      ang_dist->Fill(costheta,phi);
+
+      //Print out values
+      std::cout<<"Mass = "<<electron->Mass()<<std::endl;
+      std::cout<<"px = "<<pe.Px()<< " py = "<<pe.Py()<<" pz = "<<pe.Pz()<<std::endl;
+      std::cout<<"vx = "<<vx<< " vy = "<<vy<<" vz = "<<vz<<std::endl;
+      std::cout<<"costheta = "<<costheta<<" phi = "<<phi<<std::endl;
+      
+
+    }//If is scattering
+
+  }//end event loop
+
+  //TFile *file1 = new TFile(dir+"/Plots/histos.root","recreate");
+  TCanvas canvas("canvas");
+  vx_dist->Draw("HIST");
+  vx_dist->SetLineColor(kBlack);
+  //vx_dist->SetLineWidth
+  vx_dist->SetFillColorAlpha(kBlue,0.35);
+  vx_dist->GetXaxis()->SetTitle("v [c]");
+  vx_dist->SetStats(kFALSE);
+  //vx_dist->Write();
+  vy_dist->Draw("SAME");
+  vy_dist->SetLineColor(kBlack);
+  vy_dist->SetFillColorAlpha(kRed,0.35);
+  //vy_dist->Write();
+  vz_dist->Draw("SAME");
+  vz_dist->SetLineColor(kBlack);
+  vz_dist->SetFillColorAlpha(kGreen,0.35);
+  //vz_dist->Write();
+  //file1->Close();
+  auto legend = new TLegend(0.1,0.7,0.3,0.9);
+  //legend->SetHeader("The Legend Title","C"); // option "C" allows to center the header
+  legend->AddEntry(vx_dist,"vx");
+  legend->AddEntry(vy_dist,"vy");
+  legend->AddEntry(vz_dist,"vz");
+  legend->Draw();
+
+  std::string file_out = "Plots/vdist.pdf";
+  const char *fo = file_out.c_str();
+
+  canvas.Print(fo);
+
+  TCanvas canvas2("canvas2");
+  ang_dist->Draw("COLZ");
+  ang_dist->GetXaxis()->SetTitle("cos#theta");
+  ang_dist->GetYaxis()->SetTitle("#phi");
+  ang_dist->SetStats(kFALSE);
+  std::string file_out2 = "Plots/angdist.pdf";
+  const char *fo2 = file_out2.c_str();
+  canvas2.Print(fo2);
```

Comment:
Can you add an output file too and add the hists (and the canvas if you like) in the file as well?

Reply:
And make the printing of the canvas optional, most of the time we don't have an X terminal opened.